workflow/controller/dag.go

```diff
@@ -87,7 +87,31 @@ func (d *dagContext) assessDAGPhase(targetTasks []string, nodes map[string]wfv1.
 			retriesExhausted = false
 		}
 	}
+
 	if unsuccessfulPhase != "" {
+		if d.tmpl != nil && d.tmpl.DAG != nil {
```

Comment:
We don't need this check. DAG contexts are always instantiated with a template of type DAG

Reply:
OK